sdk/test-utils/test-utils/test/tracing/unitTest.spec.ts

```diff
@@ -0,0 +1,153 @@
+      const { tracingContext } = instrumenter.startSpan("contextTest");
+      // TODO: figure out how to be smarter about not wrapping sync functions in promise...
+      const result = await instrumenter.withContext(tracingContext, function() {
+        assert.equal(instrumenter.currentContext(), tracingContext);
+        return 42;
+      });
+
+      assert.equal(result, 42);
+      assert.notEqual(instrumenter.currentContext(), tracingContext);
+    });
+    it("sets the active context during async functions", async function() {
+      const { tracingContext } = instrumenter.startSpan("contextTest");
+      const result = await instrumenter.withContext(tracingContext, async function() {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        assert.equal(instrumenter.currentContext(), tracingContext);
+        return 42;
+      });
+      assert.equal(result, 42);
+      assert.notEqual(instrumenter.currentContext(), tracingContext);
+    });
+    it("resets the previous context after the function returns", async function() {
+      const existingContext = instrumenter.currentContext();
+      const { tracingContext } = instrumenter.startSpan("test");
+      await instrumenter.withContext(tracingContext, async function() {
+        // no-op
+      });
+      assert.equal(instrumenter.currentContext(), existingContext);
+    });
+  });
+});
+// do the same using MockCLientToTest and withSpan
+
+describe("TestInstrumenter with MockClient", function() {
+  let instrumenter: TestInstrumeter;
+  let client: MockClientToTest;
+  beforeEach(function() {
+    instrumenter = new TestInstrumeter();
+    client = new MockClientToTest();
+  });
+  describe("#startSpan", function() {
+    it("starts a span and adds to startedSpans array", function() {
+      client.mockGetMethod();
+      assert.equal(instrumenter.startedSpans.length, 1);
+      assert.equal(instrumenter.startedSpans[0].name, "MockClientToTest.mockGetMethod");
+    });
+  });
+});
+
+// or something that has upgraded to core-tracing preview.14
+export class MockClientToTest {
+  private record: Record<string, string>;
+  tracingClient: TracingClient;
+
+  constructor() {
+    this.record = {};
+    this.tracingClient = createTracingClient({
+      namespace: "Microsoft.Test",
+      packageInformation: {
+        name: "@azure/test",
+        version: "foobar"
+      }
+    });
+  }
+  // const myOperationResult = await withSpan("myClassName.myOperationName", (updatedOptions) => myOperation(updatedOptions), options);
+  async mockSetMethod() {
+    // TODO: how to pass in span options or tracing options??
+    // TODO: isn't the 2nd argument supposed to be the options for the callback??
+    const options = {
+      record: { key: "value" },
+      // TODO: without this being empty it complains - if i don't want to pass in anything
+      tracingOptions: {}
+    };
+
+    return this.tracingClient.withSpan("MockClientToTest.mockSetMethod", options, (options) => {
```

Comment:
@maorleger We don't have an implementation of "withSpan" function like we have in the `TracingClientImpl` in the @azure/core-tracing library. Where is it supposed to reside?

Reply:
Do we want to add the tracing client implementation in the test-utils library?